buffer/src/main/java/io/netty/buffer/FixedCompositeByteBuf.java

```diff
@@ -52,8 +52,12 @@
             direct = false;
         } else {
             ByteBuf b = buffers[0];
-            this.buffers = new Object[buffers.length];
-            this.buffers[0] = b;
+            if (copy) {
+                this.buffers = new ByteBuf[buffers.length];
+                System.arraycopy(buffers, 0, this.buffers, 0, buffers.length);
```

Comment:
Optional: consider using Arrays.copyOf, which I think avoids zeroing of the array.

Reply:
I actually used that in my first pass (see initial commit), but realized it uses reflection which can be avoided this way.

Unless JDK impls differ on this, it does not look like Arrays.copyOf avoids zeroing the array. I think the closest thing to that is java 9's `Unsafe.allocateUninitializedArray()`, but that's only for primitives.

Reply:
JDK has an instrinsic for Arrays.copyOf (and copyOfRange).  Protobuf uses it depending on if running Android or not:

https://github.com/google/protobuf/blob/9021f623e1420f513268a01a5ad43a23618a84ba/java/core/src/main/java/com/google/protobuf/ByteString.java#L94

Reply:
Interesting! Thanks for schooling me @carl-mastrangelo, I will change it to use this.